lib/ansiblelint/utils.py

```diff
@@ -231,7 +231,10 @@ def _include_children(basedir, k, v, parent_type):
 
     result = path_dwim(basedir, args[0])
     if not os.path.exists(result) and not basedir.endswith('tasks'):
-        result = path_dwim(os.path.join(basedir, '..', 'tasks'), v)
+        tasks_path = basedir.rsplit('tasks/')[0] + 'tasks/'
```

Comment:
When working with paths, you should typically use `os.path.split()` instead of normal `str.split()`.

Reply:
I looked at this today but I don't know if `os.path.split()` makes the code more robust:
```python
tasks_path = basedir
while not tasks_path[0].endswith('tasks'):
   tasks_path = os.path.split(tasks_path)
# Append slash since head of os.path.split has it stripped
tasks_path += '/'
```
(didn't test this approach)

Does it make the code better?

Reply:
The rsplit approach seems more readable/maintainable. Although for joining, maybe `os.path.join()` could be used over `+`.
```
tasks_root = basedir.rsplit('tasks/')[0]
result = path_dwim(os.path.join(tasks_root, 'tasks'), args[0])
```